searches/binary_search.go

```diff
@@ -18,12 +18,12 @@ func iterBinarySearch(array []int, target int, lowIndex int, highIndex int) int
 	startIndex := lowIndex
 	endIndex := highIndex
 	var mid int
-	for startIndex < endIndex {
-		mid = int(startIndex + (endIndex - startIndex))
+	for startIndex <= endIndex {
+		mid = int(startIndex + (endIndex-1)/2)
```

Comment:
- If highIndex is len(arrayName)-1, then why subtract 1 from the endIndex?

Reply:
You are right, my mistake.